typing/env.ml

```diff
@@ -426,11 +428,15 @@ let read_pers_struct check modname filename =
            } in
   if ps.ps_name <> modname then
     error (Illegal_renaming(modname, ps.ps_name, filename));
+
   List.iter
     (function
         | Rectypes ->
             if not !Clflags.recursive_types then
               error (Need_recursive_types(ps.ps_name, !current_unit))
+        | Unsafe_string ->
+            if Config.safe_string then
+              error (Depend_on_unsafe_string_unit (ps.ps_name, !current_unit));
```

Comment:
Would this break existing code not using the new configure switch, if safe and unsafe modules are mixed? That seems like it could break a lot of code, even if it should have been broken already.